src/net_processing.cpp

```diff
@@ -4767,28 +4766,24 @@ bool PeerManagerImpl::ProcessMessages(CNode* pfrom, std::atomic<bool>& interrupt
         }
     }
 
+    bool has_more_orphans;
     {
-        LOCK2(cs_main, g_cs_orphans);
-        if (!peer->m_orphan_work_set.empty()) {
-            ProcessOrphanTx(peer->m_orphan_work_set);
-        }
+        LOCK(cs_main);
+        has_more_orphans = ProcessOrphanTx(*peer);
     }
 
     if (pfrom->fDisconnect)
         return false;
 
+    if (has_more_orphans) return true;
```

Comment:
Just wanted to note that you moved this check above the `m_getdata_requests.empty()` check. Any reason behind that? (Don't see how this changes anything, so feel free to just mark as resolved if there was no reason.)

Reply:
It avoids taking the `m_getdata_requests_mutex` without changing the behaviour otherwise.